bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Control.java

```diff
@@ -3631,6 +3656,20 @@ boolean setRadioSelection (boolean value) {
  */
 public void setRedraw (boolean redraw) {
 	checkWidget ();
+
+	/*
+	 * Some embedded applications like webview2 (Edge), Excel, etc
+	 * have a renderer that will draw the whole application white
+	 * when setting redraw to false and will repaint it completely
+	 * when setting it to true. This causes flickering, which is why
+	 * turning redraw on/off is disabled in these cases.
+	 *
+	 * https://github.com/eclipse-platform/eclipse.platform.swt/issues/1122
+	 */
+	if(hasEmbeddedChildren()) {
+		return;
+	}
```

Comment:
This should only be done when `redraw==false` shouldn't it? Otherwise you may set redrawing to false when no embedded control is open, then open an embedded control, and redrawing will then never be reenabled.

Reply:
Done.